nixos/modules/services/networking/imaginary.nix

```diff
@@ -0,0 +1,100 @@
+{ lib, config, pkgs, ... }:
+
+let
+  inherit (lib) mdDoc mkEnableOption mkIf mkOption types;
+
+  cfg = config.services.imaginary;
+in {
+  options.services.imaginary = {
+    enable = mkEnableOption (mdDoc "imaginary image processing microservice");
+
+    settings = mkOption {
+      description = mdDoc ''
+        Command line arguments passed to the imaginary executable, stripped of
+        the prefix `-`. See upstream's
+        [README](https://github.com/h2non/imaginary#command-line-usage) for all
+        options.
+      '';
+      type = types.submodule {
+        freeformType = with types; attrsOf (oneOf [
+          bool
+          int
+          (nonEmptyListOf str)
+          str
+        ]);
+
+        options = {
+          a = mkOption {
```

Comment:
I'm also not super happy with it but I think this would be against the spirit of https://github.com/NixOS/rfcs/pull/42.
However we could file a PR upstream that introduces e.g. `-address` as an alias for `-a`.

Reply:
I guess we could just add these options outside of `settings`.